product_customer_code_purchase/models/purchase.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# © 2016 Nicola Malcontenti - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import models, fields, api
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    product_customer_code = fields.Char(
+        'Product Customer Code', compute='_compute_customer_code')
+
+    @api.one
+    def _compute_customer_code(self):
+        code = u''
+        product_customer_code_obj = self.env['product.customer.code']
+        partner = self.partner_id
+        product = self.product_id
+        if product and partner:
+            code_ids = product_customer_code_obj.search([
+                ('product_id', '=', product.id),
+                ('partner_id', '=', partner.id)], limit=1)
+            if code_ids:
+                code = product_customer_code_obj.browse(
```

Comment:
In new api, I think, you can do : code = code_ids.product_code or ''
search give you a browse_record